network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -1121,4 +1305,256 @@ public Optional<Resource<IdentifiableAttributes>> getIdentifiable(UUID networkUu
+            }
+        }
+        return result;
+    }
+
+    private List<TemporaryCurrentLimitAttributes> getTemporaryLimitsFromDanglingLine(UUID networkUuid, List<Resource<DanglingLineAttributes>> resources) {
+        ArrayList<TemporaryCurrentLimitAttributes> result = new ArrayList<>();
+        if (!resources.isEmpty()) {
+            for (Resource<DanglingLineAttributes> resource : resources) {
+                DanglingLineAttributes equipment = resource.getAttributes();
+
+                // currentLimits
+                result.addAll(internalPrepareTemporaryLimits(networkUuid, resource.getVariantNum(), 1,
+                        resource.getId(), DANGLING_LINE, TemporaryLimitType.CURRENT_LIMIT, equipment.getCurrentLimits()));
+
+                // apparentPowerLimits
+                result.addAll(internalPrepareTemporaryLimits(networkUuid, resource.getVariantNum(), 1,
+                        resource.getId(), DANGLING_LINE, TemporaryLimitType.APPARENT_POWER_LIMIT, equipment.getApparentPowerLimits()));
+
+                // activePowerLimits
+                result.addAll(internalPrepareTemporaryLimits(networkUuid, resource.getVariantNum(), 1,
+                        resource.getId(), DANGLING_LINE, TemporaryLimitType.ACTIVE_POWER_LIMIT, equipment.getActivePowerLimits()));
+            }
+        }
+        return result;
+    }
+
+    protected <T extends LimitSelector & IdentifiableAttributes> void insertTemporaryLimitsInEquipments(List<Resource<T>> equipments, List<TemporaryCurrentLimitAttributes> temporaryLimits) {
+        // Some equipments can have temporary limits.
+        // Those limits are not in the database table representation of the equipment, they are in a different
+        // table : "temporarylimit".
+        // We need to complete the equipments we get from the database by searching the corresponding temporary limits
+        // and inserting those limits inside the corresponding equipments.
+        // The choosen algorithm to do this is to retrieve all the temporary limits for a networkUuid, variantNum and
+        // side, then check each limit's equipment ID to see if it is the same ID as an equipment's.
+        // If it is, then it means the temporary limit belongs to the equipment.
+
+        if (!temporaryLimits.isEmpty() && !equipments.isEmpty()) {
+            // For each equipment, we will check if there are temporary limits with the corresponding equipment IDs.
+            // If there is, then we add the temporary limit to the equipment's temporaryLimits
+            // First, we put the temporary limits in a hashmap, with the map's key equals to the equipmentID.
+            // Then, for each equipment, we will load the corresponding temporary limits from the hashmap.
+            HashMap<String, List<TemporaryCurrentLimitAttributes>> hashTemporaryLimits = new HashMap<>(equipments.size());
+            for (TemporaryCurrentLimitAttributes temporaryLimitResource : temporaryLimits) {
```

Comment:
Why do we need two loops here ?
Why not looping over `TemporaryLimits`, then get the corresponding `equipment` then `insertTemporaryLimitsInEquipment`. 
Must check the complexity of equipments List access by ID ?
Maybe take a hashmap of equipment (keys=id) as argument here ?

Reply:
After discussion, this optimization could be discussed later, in another US.